### DISCUSSION TITLE
AverageExpression() for TF-IDF normalized scATAC data

### DISCUSSION THREAD

**Discussion by danielcgingerich** [stuart-lab/signac#740] [Q&A] [answered]
What is the best way to calculate average expression of scATAC data?  Should I just average the raw counts?  Seurat exponentiates values prior to taking the mean to convert log transformed values to raw counts.  ATAC data cant simply be exponentiated to convert to raw counts because the log1p is done on the tf-idf values.  So how would you go about getting the average expression of ATAC data?  

Is there a problem with just averaging the raw count values? 

Best, 

Dan

**Comment by timoast** [comment, marked as answer]
Yes you need to run `AverageExpression()` with `slot="counts"` to average the count data. If `slot="data"` (the default for `AverageExpression()`), the results will not be accurate.

**Comment by chrarnold** [comment]
I have a similar problem currently, trying to find the best way to extract average accessilibity for scATAC data that has been TF-IDF normalized.

If using the `counts` slot as suggested by @timoast , wouldnt there be a potential bias for cells with high coverage. Cells of a particular cell type with distinct accessibility profiles could therefore bias the mean value, right? 

Are there alternative solutions? I also saw this here, but not whether that really works well and it appears "involved": #318 

Lastly, maybe the alternative ways of how the TF-IDF is done exactly may work better for generating (depth-unaffected) counts to generate a mean value?
